nifi-nar-bundles/nifi-asn1-bundle/nifi-asn1-services/src/main/java/org/apache/nifi/jasn1/JASN1Reader.java

```diff
@@ -134,10 +135,35 @@ public class JASN1Reader extends AbstractConfigurableComponent implements Record
         .required(false)
         .build();
 
+    private static final PropertyDescriptor DO_ADDITIONAL_PREPROCESSING = new PropertyDescriptor.Builder()
+        .name("do-additional-preprocessing")
+        .displayName("Do Additional Preprocessing")
+        .description("When set, NiFi will do additional preprocessing steps that creates modified versions of the provided ASN files," +
+                " removing unsupported features in a way that makes them less strict but otherwise should still be compatible with incoming data." +
+                " The original files will remain intact and new ones will be created with the same names in the directory defined in the 'Additional Preprocessing Output Directory' property." +
+                " For more information about these additional preprocessing steps please see Additional Details - Additional Preprocessing.")
+        .allowableValues("true", "false")
+        .required(true)
+        .defaultValue("false")
+        .build();
+
+    private static final PropertyDescriptor ADDITIONAL_PREPROCESSING_OUTPUT_DIRECTORY = new PropertyDescriptor.Builder()
+        .name("additional-preprocessing-output-directory")
+        .displayName("Additional Preprocessing Output Directory")
```

Comment:
Following the above naming suggestion, recommend renaming this property to `Schema Preparation Directory`.